common/src/main/java/com/microsoft/identity/common/migration/ISharedPrefsFileManagerReencrypter.java

```diff
@@ -0,0 +1,143 @@
+
+/**
+ * Interface describing an object that can reencrypt instances of
+ * {@link ISharedPreferencesFileManager}.
+ */
+public interface ISharedPrefsFileManagerReencrypter {
+
+    /**
+     * The object to which this class delegates reencryption of the
+     * {@link ISharedPreferencesFileManager}.
+     */
+    interface IStringEncrypter {
+        String encrypt(String input) throws Exception;
+    }
+
+    /**
+     * The object to which this class delegates decryption of the input
+     * {@link ISharedPreferencesFileManager}.
+     */
+    interface IStringDecrypter {
+        String decrypt(String input) throws Exception;
+    }
+
+    /**
+     * Encapsulates error handling switches for controlling reencryption.
+     */
+    class ReencryptionParams {
+
+        private final boolean mAbortOnError;
+        private final boolean mEraseEntryOnError;
+        private final boolean mEraseAllOnError;
+
+        /**
+         * Constructs a new {@link ReencryptionParams}.
+         *
+         * @param abortOnError      True if the operation should abort upon errors.
+         * @param eraseEntryOnError True, if the operation should delete the entry that caused the error.
+         * @param eraseAllOnError   True, if the operation should delete all entries if an error occurs.
+         */
+        public ReencryptionParams(final boolean abortOnError,
+                                  final boolean eraseEntryOnError,
+                                  final boolean eraseAllOnError) {
+            mAbortOnError = abortOnError;
+            mEraseEntryOnError = eraseEntryOnError;
+            mEraseAllOnError = eraseAllOnError;
+        }
+
+        /**
+         * Gets the abortOnError value.
+         *
+         * @return True, if the operation should abort upon error.
+         */
+        boolean abortOnError() {
+            return mAbortOnError;
+        }
+
+        /**
+         * Gets the eraseEntryOnError value.
+         *
+         * @return True, if the operation should erase the problematic entry upon error.
+         */
+        boolean eraseEntryOnError() {
+            return mEraseEntryOnError;
+        }
+
+        /**
+         * Gets the eraseAllOnError value.
+         *
+         * @return True, if the operation should erase all entries upon error.
+         */
+        boolean eraseAllOnError() {
+            return mEraseAllOnError;
+        }
+    }
+
+    /**
+     * Performs reencryption of the provided {@link ISharedPreferencesFileManager}, delegating to
+     * the suppplied {@link IStringEncrypter} and {@link IStringDecrypter} to perform content
+     * transformations.
+     * <p>
+     * Please note: this method does not lock the underlying store during reencryption. Users of
+     * this API are advised to ensure the designated store is not mutated during the reencryption
+     * process otherwise undefined behavior/results may occur.
```

Comment:
nit:
Should this be specified in the methodName..
reencryptWithoutLock?

Reply:
This is a good suggestion to be explicit - I will decline it for now such that this PR remains a single cherry-pickable commit and due to the fact that this method is _essentially_ private API as it will be called internally from OneAuth. The fact that this storage cannot be locked will be captured in two places that hopefully provide adequate warning/context for those executing this code: first, here via javadoc, and again via the OneAuth migration documentation that will be updated to assist developers transitioning from ADAL -> OneAuth